### PR TITLE
fix(sync): Extract error message from gRPC error

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Related to https://github.com/cloudquery/cloudquery/issues/4707.
Errors returned from a gRPC call have both a status and a message. We don't really use the status, but when we wrap the error it always gets printed, see:
https://github.com/grpc/grpc-go/blob/eeb9afa1f6b6388152955eeca8926e36ca94c768/internal/status/status.go#L140

We can easily extract just the message and print only that.
Before:
```
Error: failed to sync source aws: failed to fetch resources from stream: rpc error: code = Unknown desc = failed to sync resources: failed to create execution client for source plugin aws: no enabled accounts instantiated
```
After:
```
Error: failed to sync source aws: failed to sync resources: failed to create execution client for source plugin aws: no enabled accounts instantiated
```

We should do this for all gRPC calls, but I'll do that in follow up PRs so I can test it

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: clients/source.go [hunk 1]
```diff
@@ -307,15 +307,15 @@ func (c *SourceClient) Sync2(ctx context.Context, spec specs.Source, res chan<-
...
-		return fmt.Errorf("failed to call Sync: %w", err)
+		return fmt.Errorf("failed to call Sync: %s", extractErrorMessage(err))
 	}
 	for {
 		r, err := stream.Recv()
 		if err != nil {
 			if err == io.EOF {
 				return nil
 			}
-			return fmt.Errorf("failed to fetch resources from stream: %w", err)
```
Comment (erezrokah) [inline clients/source.go:318]: `failed to fetch resources from stream` is incorrect (similar to https://github.com/cloudquery/plugin-sdk/issues/462#issue-1476833829), as error can be anything returned from https://github.com/cloudquery/plugin-sdk/blob/cdcef5d9256ee7d43336e557d1c4db3eda2ee46b/internal/servers/source.go#L93 so not necessarily means something with the stream

**Review by disq — APPROVED** [review]

**Review by yevgenypats — CHANGES_REQUESTED** [review]
What is the value of extracting it rather than printing it as is (without more code) with the `grpc=` given this error comes from grpc so basically we are hiding more information just because the string is a bit long?

**Comment by erezrokah** [comment]
> given this error comes from grpc so basically we are hiding more information just because the string is a bit long?

This is not a gRPC error. This is an application level error and we almost always print `rpc error: code = Unknown desc =` which has no value. See also bullet 2. in https://github.com/cloudquery/plugin-sdk/pull/485#issue-1489972263 and issue https://github.com/cloudquery/plugin-sdk/issues/462

**Comment by yevgenypats** [comment]
> > given this error comes from grpc so basically we are hiding more information just because the string is a bit long?
> 
> This is not a gRPC error. This is an application level error and we almost always print `rpc error: code = Unknown desc =` which has no value. See also bullet 2. in [#485 (comment)](https://github.com/cloudquery/plugin-sdk/pull/485#issue-1489972263) and issue #462.
> 
> Also explained in [#477 (comment)](https://github.com/cloudquery/plugin-sdk/pull/477#discussion_r1043569021)

Right now we don't have a way to send application level errors. We can implement this type of behaviour and add a field to the protocol but it's not the right place to do it over grpc errors.

**Comment by erezrokah** [comment]
> We can implement this type of behaviour and add a field to the protocol but it's not the right place to do it over grpc errors.

Agree we should not do it as a part of gRPC errors, hence https://github.com/cloudquery/plugin-sdk/pull/487. We don't need to add anything to the protocol as `errdetails` already has all the things generated.

This PR doesn't change error handling, its meant to only remove the status print `rpc error: code = Unknown desc` as we're not using it anywhere nor it's useful to the user, pending a better solution.

**Comment by erezrokah** [comment]
Closing since I think the fixes in https://github.com/cloudquery/plugin-sdk/pull/530 should be enough for now. We can re-visit this in the future

**Pull request closed without merging**